### PR TITLE
Follows the NGINX Breaking Changes suggestion

### PR DESCRIPTION
**Pull request by carlosasj**
On the [Compatibility > Breaking changes](https://docs.mattermost.com/administration/changelog.html#compatibility) section, they asked us to replace one line in the NGINX file, according to [documentation](https://docs.mattermost.com/install/install-ubuntu-1404.html#configuring-nginx-as-a-proxy-for-mattermost-server).

I updated the whole file to be more similar with the documentation.

**Please, pay special attention to the SSL version of the NGINX file, since I'm not familiar with SSL configs on NGINX, so it's more susceptible to contain errors.**

### CODE REVIEW

File: web/mattermost [hunk 1]
```diff
@@ -6,17 +6,34 @@ map $http_x_forwarded_proto $proxy_x_forwarded_proto {
...
 
         client_max_body_size 50M;
-        proxy_set_header Upgrade $http_upgrade;
-        proxy_set_header Connection "upgrade";
+        proxy_set_header Connection "";
         proxy_set_header Host $http_host;
         proxy_set_header X-Real-IP $remote_addr;
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-        proxy_set_header X-Forwarded-Proto $proxy_x_forwarded_proto;
+        proxy_set_header X-Forwarded-Proto $scheme;
```
Comment (pichouk) [inline web/mattermost]: You should conserve the `$proxy_x_forwarded_proto` here.
You can read #58 to get more information, but using `$scheme` here can lead to some issue on Mattermost.

File: web/mattermost-ssl [hunk 2]
```diff
@@ -20,18 +20,37 @@ server {
...
 
         client_max_body_size 50M;
-        proxy_set_header Upgrade $http_upgrade;
-        proxy_set_header Connection "upgrade";
+        proxy_set_header Connection "";
         proxy_set_header Host $http_host;
         proxy_set_header X-Real-IP $remote_addr;
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-        proxy_set_header X-Forwarded-Proto $proxy_x_forwarded_proto;
+        proxy_set_header X-Forwarded-Proto $scheme;
```
Comment (pichouk) [inline web/mattermost-ssl]: Same here, should keep `$proxy_x_forwarded_proto`.

**Comment by carlosasj** [comment]
@pichouk I did your suggestions. Thanks for your review 😄

**Comment by pichouk** [comment]
Looks good to me like this :)

**Pull request merged**